File: php/src/Builder/SQLBuilder/SqlQuery.php
<?php

namespace Ibrahim13\PdPhp\Builder\SqlBuilder;



class SqlQuery 
{
	
	public	$table;

	public $orderBy;

	public $conditions = [];

	public $fields = [];

	public $limit;



public function __toString() {
    // Handle fields
    $fields = empty($this->fields) ? '*' : implode(', ', (array) $this->fields);

    // Start query
    $query = "SELECT {$fields} FROM {$this->table}";

    // Handle WHERE clause
    if (!empty($this->conditions)) {
        $query .= " WHERE " . implode(' AND ', $this->conditions);
    }

    // Handle ORDER BY clause
    if (!empty($this->orderBy)) {
        $query .= " ORDER BY {$this->orderBy}";
    }

    // Handle LIMIT clause
    if (!empty($this->limit)) {
        $query .= " LIMIT {$this->limit}";
    }

    return $query;
}

	
}

File: python/builder/sql_query.py



class SqlQuery:

    def __init__(self):
        
        self.table = ''
        self.fields = []
        self.conditions = []
        self.limit = ''
        self.orderBy = ''


    def __str__(self) -> str:
       
       fields = "*" if not self.fields else ", ".join(self.fields)
       
       sql =f"SELECT {fields} FROM {self.table}"    
       
       if self.conditions:
           sql += " WHERE " + "AND".join( self.conditions )
        
       if self.limit :
           sql += f" LIMIT {self.limit}"
        
       if self.orderBy :
           sql += f" ORDER BY {self.orderBy}"

       return sql